submissions/YuliiaHordiichuk/frogger/app.js

```diff
@@ -0,0 +1,128 @@
+const canvasWidth = 505;  
+const startX = 200; 
+const startY = 400; 
+const stepX = 100; 
+const stepY = 85; 
+const minX = 0; 
+const maxX = minX + 4 * stepX; 
+const minY = -25; 
+const maxY = startY;  
+const startEnemyX = -110 + Math.floor((Math.random()) * (-50));
+
+const Enemy = function(x, y) {
+    this.x = x; 
+    this.y = y; 
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += dt * 200; 
+    this.checkEnemyLocation(); 
+    this.checkCollision(); 
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkEnemyLocation = function() {
+    if (this.x > canvasWidth) {
+        this.x = startEnemyX;  
+    }
+}; 
+
+Enemy.prototype.checkCollision = function() {
+    if(Math.abs(player.x - this.x) < 80 && this.y === player.y) {
+        setTimeout(() => {
+            alert ('You lost'); 
+            player.restart();
+        },0) 
+    };
```

Comment:
Classes shouldn't refer to any global variables directly.
Exclusions are global constants and, in this particular exercise, entities provided by the engine (`Resources`).

Reply:
thanks for the comments, done